test/functional/feature_framework_startup_failures.py

```diff
@@ -0,0 +1,127 @@
+        self._verify_startup_failure(
+            TestWrongRpcPortStartupFailure,
+            r"AssertionError: \[node 0\] Unable to connect to bitcoind after \d+s \(ignored errors: {[^}]*'OSError \w+'?: \d+[^}]*}, latest error: \w+\([^)]+\)\)"
+        )
+
+        self.log.info("Verify startup failure due to invalid arg results in only one exception.")
+        self._verify_startup_failure(
+            TestInitErrorStartupFailure,
+            r"FailedToStartError: \[node 0\] bitcoind exited with status 1 during initialization\. Error: Error parsing command line arguments: Invalid parameter -nonexistentarg"
+        )
+
+        self.log.info("Verify start() then stop_node() on a node without wait_for_rpc_connection() in between raises an assert.")
+        self._verify_startup_failure(
+            TestStartStopStartupFailure,
+            r"AssertionError: \[node 0\] Should only call stop_node\(\) on a running node after wait_for_rpc_connection\(\) succeeded\. Did you forget to call the latter after start\(\)\? Not connected to process: \d+"
+        )
+
+class InternalTestMixin:
+    def add_options(self, parser):
+        # Just here to silence unrecognized argument error, we actually read the value in the if-main at the bottom.
+        parser.add_argument("--internal_test", dest="internal_never_read", help="ONLY TO BE USED WHEN TEST RELAUNCHES ITSELF")
+
+class TestWrongRpcPortStartupFailure(InternalTestMixin, BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        # Lower the timeout so we don't wait that long.
+        self.rpc_timeout = max(3, self.options.timeout_factor)
+        # Override RPC port to something TestNode isn't expecting so that we
+        # are unable to establish an RPC connection.
+        self.extra_args = [[f"-rpcport={rpc_port(2)}"]]
+
+    def run_test(self):
+        assert False, "Should have failed earlier during startup."
+
+class TestInitErrorStartupFailure(InternalTestMixin, BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.extra_args = [["-nonexistentarg"]]
+
+    def run_test(self):
+        assert False, "Should have failed earlier during startup."
+
+class TestStartStopStartupFailure(InternalTestMixin, BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.add_nodes(self.num_nodes, self.extra_args)
+        self.nodes[0].start()
+        self.nodes[0].stop_node() # This should raise an exception
+        assert False, "stop_node() should raise an exception when we haven't called wait_for_rpc_connection()"
+
+    def run_test(self):
+        assert False, "Should have failed earlier during startup."
+
+
+if __name__ == '__main__':
+    if not any('--internal_test=' in arg for arg in sys.argv):
+        FeatureFrameworkStartupFailures(__file__).main()
+    else:
+        for arg in sys.argv:
+            if arg[:16] == '--internal_test=':
```

Comment:
nit: can probably be `if arg.startswith('--internal_test='):`
or walrus to the rescue again:
```python
if arg.startswith(prefix := '--internal_test='):
    class_name = arg[len(prefix):]
```

Reply:
In commit "Drop or Squash? - Break out startup failures into distinct classes" (9e0fede6a6d608ff307b48b9a326c6d467b410cf)

I think could also simplify the `sys.modules` code and write `globals()[class_name](__file__).main()`